File: aspnet-in-practice/15/VB/AuthorizationModule/App_Code/AuthorizationModule.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Configuration
Imports System.IO
Imports System.Web.UI

Namespace ASPNET4InPractice
	Public Class AuthorizationModule
		Implements IHttpModule
		Public Sub Dispose() Implements IHttpModule.Dispose
			' nothing to do
		End Sub

		Public Sub Init(ByVal context As HttpApplication) Implements IHttpModule.Init
			AddHandler context.AuthorizeRequest, AddressOf OnAuthorizeRequest
		End Sub

		Private Sub OnAuthorizeRequest(ByVal sender As Object, ByVal e As EventArgs)
			Dim app As HttpApplication = DirectCast(sender, HttpApplication)

			' probably you'll implement something more useful than this :)
			If DateTime.Now.Hour >= 17 Then
				app.Context.Response.StatusCode = 401
			End If
		End Sub

	End Class
End Namespace

File: aspnet-in-practice/15/CS/AuthorizationModule/App_Code/AuthorizationModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Configuration;
using System.IO;
using System.Web.UI;

namespace ASPNET4InPractice
{
	public class AuthorizationModule : IHttpModule
	{
		public void Dispose()
		{
			// nothing to do
		}

		public void Init(HttpApplication context)
		{
			context.AuthorizeRequest += new EventHandler(OnAuthorizeRequest);
		}

		void OnAuthorizeRequest(object sender, EventArgs e)
		{
			HttpApplication app = (HttpApplication)sender;

			// probably you'll implement something more useful than this :)
			if (DateTime.Now.Hour >= 17)
				app.Context.Response.StatusCode = 401;
		}

	}
}